### PR TITLE
[CBRD-20631] fix recovery issues from file manager redesign

### PR DESCRIPTION
**Pull request by ribram**
* always initialize new permanent pages before logging page allocation. on undo the page is deallocated and it is expected to be fully allocated (its page type different from PAGE_UNKNOWN)
* remove file_alloc_and_init. file_alloc receives init arguments by default.
* fix log_recovery_finish_postpone
  - remove obsolete start_postpone_run_lsa argument from log_recovery_find_first_postpone (was still used by mistake)
  - fix rollbacking interruped logical run postpone system operation. lastparent_lsa is set to whichever is last: LSA of system op/transaction start postpone.
  - fix log_get_next_nested_top. the LSA ranges that must be skipped should not include the LSA of end system op log record if it is logical run postpone.

### CODE REVIEW

File: src/storage/overflow_file.c [hunk 1]
```diff
@@ -522,22 +511,15 @@ overflow_update (THREAD_ENTRY * thread_p, const VFID * ovf_vfid, const VPID * ov
...
 
-      if (isnewpage)
-	{
-	  log_append_redo_data (thread_p, RVOVF_NEWPAGE_INSERT, &addr, copy_length + hdr_length, addr.pgptr);
-	}
-      else
-	{
-	  log_append_redo_data (thread_p, RVOVF_PAGE_UPDATE, &addr, copy_length + hdr_length, (char *) addr.pgptr);
-	}
+      log_append_redo_data (thread_p, RVOVF_PAGE_UPDATE, &addr, copy_length + hdr_length, (char *) addr.pgptr);
```
Comment (eseokoh) [inline src/storage/overflow_file.c]: Please look at `la_get_recdes`:

```c
  if (*rcvindex == RVOVF_CHANGE_LINK)
    {
      /* if overflow page update */
      error = la_get_overflow_recdes (lrec, logs, recdes, RVOVF_PAGE_UPDATE);
      recdes->type = REC_BIGONE;
    }
  else if (recdes->type == REC_BIGONE)
    {
      /* if overflow page insert */
      error = la_get_overflow_recdes (lrec, logs, recdes, RVOVF_NEWPAGE_INSERT);
    }
  else if (*rcvindex == RVHF_INSERT && recdes->type == REC_ASSIGN_ADDRESS)
    {
      error = la_get_next_update_log (lrec, pg, &logs, &rec_type, &recdes->data, &recdes->length);
      if (error == NO_ERROR)
        {
          recdes->type = *(INT16 *) (rec_type);
          if (recdes->type == REC_BIGONE)
            {
              error = la_get_overflow_recdes (lrec, logs, recdes, RVOVF_NEWPAGE_INSERT);
            }
        }
    }
```

It expects `RVOVF_NEWPAGE_INSERT` log to trace overflow insertion. 
I've thought about replacing it as `RVOVF_PAGE_UPDATE` but I'm not sure. 
Maybe it is better to leave a dummy log record to help log applier.

File: src/transaction/log_manager.c [hunk 2]
```diff
@@ -8289,6 +8291,8 @@ log_get_next_nested_top (THREAD_ENTRY * thread_p, LOG_TDES * tdes, LOG_LSA * sta
       if (log_rec->type == LOG_SYSOP_END)
 	{
 	  /* Read the DATA HEADER */
+	  LOG_LSA prev_tran_lsa = log_rec->back_lsa;
```
Comment (eseokoh) [inline src/transaction/log_manager.c]: Which is better between `back_lsa` and `prev_tranlsa`?

File: src/storage/btree_load.c [hunk 3]
```diff
@@ -1885,28 +1882,19 @@ btree_load_new_page (THREAD_ENTRY * thread_p, const BTID * btid, BTREE_NODE_HEAD
...
-  alignment = BTREE_MAX_ALIGN;
-
-  spage_initialize (thread_p, *page_new, UNANCHORED_KEEP_SEQUENCE, alignment, DONT_SAFEGUARD_RVSPACE);
-
-  addr.vfid = &btid->vfid;
-  addr.offset = -1;		/* No header slot is initialized */
-  addr.pgptr = *page_new;
-
-  log_append_redo_data (thread_p, RVBT_GET_NEWPAGE, &addr, sizeof (alignment), &alignment);
+  (void) pgbuf_check_page_ptype (thread_p, *page_new, PAGE_BTREE);
```
Comment (eseokoh) [inline src/storage/btree_load.c]: I think it is redundant.

File: src/storage/extendible_hash.c [hunk 4]
```diff
@@ -1084,20 +1085,17 @@ ehash_create_helper (THREAD_ENTRY * thread_p, EHID * ehid_p, DB_TYPE key_type, i
...
-  dir_page_p = pgbuf_fix (thread_p, &dir_vpid, NEW_PAGE, PGBUF_LATCH_WRITE, PGBUF_UNCONDITIONAL_LATCH);
   if (dir_page_p == NULL)
     {
-      ASSERT_ERROR ();
+      assert_release (false);
       goto exit_on_error;
     }
-
-  pgbuf_set_page_ptype (thread_p, dir_page_p, PAGE_EHASH);
+  pgbuf_check_page_ptype (thread_p, dir_page_p, PAGE_EHASH);
```
Comment (eseokoh) [inline src/storage/extendible_hash.c]: Isn't it redundant?

File: src/storage/extendible_hash.c [hunk 5]
```diff
@@ -1487,22 +1485,21 @@ ehash_insert_to_bucket_after_create (THREAD_ENTRY * thread_p, EHID * ehid_p, PAG
...
     }
-
-  bucket_page_p = ehash_fix_old_page (thread_p, &dir_header_p->bucket_file, bucket_vpid_p, PGBUF_LATCH_WRITE);
   if (bucket_page_p == NULL)
     {
+      assert_release (false);
       log_sysop_abort (thread_p);
       return ER_FAILED;
     }
+  (void) pgbuf_check_page_ptype (thread_p, bucket_page_p, PAGE_EHASH);
```
Comment (eseokoh) [inline src/storage/extendible_hash.c]: Isn't it redundant?

File: src/storage/system_catalog.c [hunk 6]
```diff
@@ -657,26 +657,19 @@ catalog_get_new_page (THREAD_ENTRY * thread_p, VPID * page_id_p, bool is_overflo
...
-  page_p = pgbuf_fix (thread_p, page_id_p, OLD_PAGE, PGBUF_LATCH_WRITE, PGBUF_UNCONDITIONAL_LATCH);
   if (page_p == NULL)
     {
-      ASSERT_ERROR ();
+      assert_release (false);
       log_sysop_abort (thread_p);
       return NULL;
     }
-
   (void) pgbuf_check_page_ptype (thread_p, page_p, PAGE_CATALOG);
```
Comment (eseokoh) [inline src/storage/system_catalog.c]: Isn't it redundant?

**Comment by eseokoh** [comment]
I'm sorry I was confused with `pgbuf_check_page_type`. I misread it is `pgbuf_set_page_ptype`. :)

File: src/storage/overflow_file.c (hunk 1 above)
Comment (ribram) [inline src/storage/overflow_file.c]: You are right, it is better to use RVOVF_NEWPAGE_INSERT. The recovery is the same, but log applier can see them differently. Probably RVOVF_UPDATE_PAGE could also be used, but it's more clear this way.

I updated the code.

**Pull request merged**